### PR TITLE
move notification builder from host to common

### PR DESCRIPTION
**Pull request by ranakan19**
Move notification Builder from host to toolchain-common

changes wrt - https://issues.redhat.com/browse/CRT-1418

### CODE REVIEW

File: pkg/usersignup/usersignup.go [hunk 1]
```diff
@@ -1,6 +1,12 @@
 package usersignup
 
 import (
+	"crypto/md5" // nolint:gosec
```
Comment (alexeykazakov) [inline pkg/usersignup/usersignup.go:4]: This change in `usersignup.go` doesn't seem to be related to the builder.

Comment (ranakan19) [reply pkg/usersignup/usersignup.go:4]: the notification_builder_test uses the changes in usersignup.go

Comment (alexeykazakov) [reply pkg/usersignup/usersignup.go:4]: OK. Then I guess you don't need to keep this code in the host operator anymore and can switch to common here: https://github.com/codeready-toolchain/host-operator/pull/629/files

Comment (ranakan19) [reply pkg/usersignup/usersignup.go:4]: done

**Review by alexeykazakov — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

File: pkg/notification/notification_builder_test.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
+package notification
+
+import (
+	"testing"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/codeready-toolchain/api/api/v1alpha1"
+	test2 "github.com/codeready-toolchain/toolchain-common/pkg/test/usersignup"
```
Comment (rajivnathan) [inline pkg/notification/notification_builder_test.go:9]: Probably better to use an alias that's related to usersignup for this. eg. `testusersignup`

**Review by rajivnathan — APPROVED** [review]
Looks good, just left 1 minor comment
